crates/uv/src/commands/tool/install.rs

```diff
@@ -47,29 +47,44 @@ pub(crate) async fn install(
     if preview.is_disabled() {
         warn_user_once!("`uv tool install` is experimental and may change without warning.");
     }
+    let from = from.unwrap_or(name.clone());
 
     let installed_tools = InstalledTools::from_settings()?;
 
+    // TODO(zanieb): Figure out the interface here, do we infer the name or do we match the `run --from` interface?
+    let from = Requirement::<VerbatimParsedUrl>::from_str(&from)?;
+    let existing_tool_entry = installed_tools.find_tool_entry(&name)?;
     // TODO(zanieb): Automatically replace an existing tool if the request differs
-    if installed_tools.find_tool_entry(&name)?.is_some() {
+    let reinstall_entry_points = if existing_tool_entry.is_some() {
         if force {
             debug!("Replacing existing tool due to `--force` flag.");
+            false
         } else {
-            writeln!(printer.stderr(), "Tool `{name}` is already installed.")?;
-            return Ok(ExitStatus::Failure);
+            match settings.reinstall {
+                Reinstall::All => {
+                    debug!("Replacing existing tool due to `--reinstall` flag.");
+                    true
+                }
+                // Do not replace the entry points unless the tool is explicitly requested
+                Reinstall::Packages(ref packages) => packages.contains(&from.name),
+                // If not reinstalling... then we're done
+                Reinstall::None => {
+                    writeln!(printer.stderr(), "Tool `{name}` is already installed.")?;
```

Comment:
Lets omit the periods from user-facing copy for consistency

Reply:
👍